src/hotspot/os/linux/os_linux.cpp

```diff
@@ -5952,6 +5957,10 @@ static int checkpoint_restore(int *shmid) {
   } while (sig == -1 && errno == EINTR);
   assert(sig == RESTORE_SIGNAL, "got what requested");
 
+  initialize_processor_count();
+  if (_cpu_to_node != NULL)
+    rebuild_cpu_to_node_map();
```

Comment:
It seems the only place where number of processors is updated, so it's not clear how safe the operation. Please add a debug option for processor count update, I propose that to be disabled by default.

Reply:
OK, moved under `-XX:+CRaCCPUCountInit`. It was crashing (asserting IIRC) for me when moving between 4 and 20 vCPUs but I was unable to reproduce it now.